### ISSUE TITLE
Sometimes Cosmetic Filtering is not applied after restart

### ISSUE THREAD

**Issue by btlechowski** [brave/brave-browser#9060]
Found when testing https://github.com/brave/brave-browser/issues/5381

Reproduced on 1.7.x, 1.8.x, 1.9.x

## Steps to Reproduce

   1. Clean profile
   2. Open `https://enpt.dict.cc/`
   3. Restart browser

## Actual result:

Cosmetic Filtering is not applied
![image](https://user-images.githubusercontent.com/34715963/78611901-3fc49080-7868-11ea-865e-c48d87de4e5b.png)

## Expected result:
Cosmetic Filtering is applied
![image](https://user-images.githubusercontent.com/34715963/78611865-2d4a5700-7868-11ea-951c-1fa609b5a9ea.png)

## Reproduces how often: 

70% repro rate

## Version/Channel Information:

Reproduced on 1.7.x, 1.8.x, 1.9.x

Brave | 1.7.86 Chromium: 80.0.3987.163 (Official Build) (64-bit)
-- | --
Revision | e7fbe071abe9328cdce4ffedac9822435fbd3656-refs/branch-heads/3987@{#1037}
OS | Ubuntu 18.04 LTS

Brave | 1.9.4 Chromium: 81.0.4044.83 (Official Build) nightly (64-bit)
-- | --
Revision | 5718ca617723806b9aacfb41108a0d911d1bde74-refs/branch-heads/4044@{#853}
OS | Windows 7 Service Pack 1 (Build 7601.24544)

cc @brave/legacy_qa @petemill @rebron

**Comment by rebron** [comment]
cc: @antonok-edm Can you take a look when you get a chance?

**Comment by SpicBeanerMower** [comment]
The problem of brave not applying cosmetic filtering goes beyond than "after restart" and I think it is related to this problem (since this is about the inbuilt cosmetic rules and not user made ones).

But when you create your own cosmetic filtering rules and you open the website or page as "new tab", the custom cosmetic rules don't apply and you have to refresh website for them to work, this problem occurs like 100% of the time, and I have tested Brave release, dev builds and now Nightly builds (currently Brave 1.9.24), all same behavior.
Opening as new window works, unless I open many links as new window, sometimes it doesn't.
And after restart sometimes it works, sometimes it doesn't, like, same behavior than "new window", like, if you have many tabs opening on start, some will apply the custom cosmetic rules and others don't.

**Comment by ShivanKaul** [comment]
@ryanbr can you take a look at the cosmetic filtering on this page? The `Advertisement` text shows up in weird places.

**With Shields:** (the text shows up right below omnibox)
<img width="367" alt="Image" src="https://github.com/user-attachments/assets/9393cc2e-166f-42af-b389-7e2174ac0ad3" />


**Without Shields:** (the ad displays, which is expected)

<img width="754" alt="Image" src="https://github.com/user-attachments/assets/fbb7d3cd-46ad-45ce-a75d-01a782745842" />

**Comment by ShivanKaul** [comment]
I don't think this is a feature limitation and is just a filter going wrong, in which case I don't think we need QA for this, but let me know @ryanbr if I'm wrong.

**Comment by ryanbr** [comment]
Apologies for the lateness in the response @btlechowski  If you're still using Brave it'll be fixed in the next 2-3hrs.
